crates/router/src/connector/cybersource.rs

```diff
@@ -33,24 +37,35 @@ impl Cybersource {
         auth: cybersource::CybersourceAuthType,
         host: String,
         resource: &str,
-        payload: &str,
+        payload: &String,
         date: OffsetDateTime,
+        http_method: services::Method,
     ) -> CustomResult<String, errors::ConnectorError> {
         let cybersource::CybersourceAuthType {
             api_key,
             merchant_account,
             api_secret,
         } = auth;
-
-        let headers_for_post_method = "host date (request-target) digest v-c-merchant-id";
+        let is_post_method = matches!(http_method, services::Method::Post);
+        let headers = vec![
+            "host date (request-target) ",
+            if is_post_method { "digest " } else { "" },
+            "v-c-merchant-id",
+        ]
+        .join("");
+        let request_target = if is_post_method {
+            format!(
+                "(request-target): post {resource}\n\
+                digest: SHA-256={payload}\n"
```

Comment:
```suggestion
                "(request-target): post {resource}\ndigest: SHA-256={payload}\n"
```

Since this is a small enough string literal, you can keep it on a single line.